### PR TITLE
[search] BookmarksProcessor is integrated in Processor.

### PR DESCRIPTION
**Pull request by ygorshenin**
@rokuz, feel free to review only SearchAPI.

### CODE REVIEW

File: search/search_params.hpp [hunk 1]
```diff
@@ -20,6 +20,7 @@ class Tracer;
 
 struct SearchParams
 {
+  static size_t const kDefaultNumBookmarksResults = 1000;
```
Comment (mpimenov) [inline search/search_params.hpp]: s/BookmarksResults/ResultsBookmarks/ for consistency

Comment (ygorshenin) [reply search/search_params.hpp]: But what is the right naming? From my point of view the current naming is better.

File: map/search_api.cpp [hunk 2]
```diff
@@ -76,6 +80,52 @@ void AppendBookmarkIds(vector<df::MarkID> const & marks, vector<bookmarks::Id> &
...
+    }
+    else
+    {
+      ASSERT_EQUAL(m_status, BookmarksSearchParams::Status::InProgress, ());
+    }
+
+    auto const & rs = results.GetBookmarksResults();
+    ASSERT_LESS_OR_EQUAL(m_lastResultsSize, rs.size(), ());
+    for (; m_lastResultsSize < rs.size(); ++m_lastResultsSize)
+      m_results.push_back(BookmarkIdToMarkID(rs[m_lastResultsSize].m_id));
```
[the remark this answers is not in the document]
Comment (ygorshenin) [reply map/search_api.cpp]: Done

File: map/search_api.cpp [hunk 3]
```diff
@@ -76,6 +80,52 @@ void AppendBookmarkIds(vector<df::MarkID> const & marks, vector<bookmarks::Id> &
   result.reserve(result.size() + marks.size());
   transform(marks.begin(), marks.end(), back_inserter(result), MarkIDToBookmarkId);
 }
+
+
```
[the remark this answers is not in the document]
Comment (ygorshenin) [reply map/search_api.cpp]: No, no need.

File: map/search_api.cpp [hunk 4]
```diff
@@ -76,6 +80,52 @@ void AppendBookmarkIds(vector<df::MarkID> const & marks, vector<bookmarks::Id> &
...
+      }
+    }
+    else
+    {
+      ASSERT_EQUAL(m_status, BookmarksSearchParams::Status::InProgress, ());
+    }
+
+    auto const & rs = results.GetBookmarksResults();
+    ASSERT_LESS_OR_EQUAL(m_lastResultsSize, rs.size(), ());
+    for (; m_lastResultsSize < rs.size(); ++m_lastResultsSize)
```
[the remark this answers is not in the document]
Comment (ygorshenin) [reply map/search_api.cpp]: I'd prefer to keep this as is. Or, even better, to simplify this via a temporary variable.

File: map/search_api.cpp [hunk 5]
```diff
@@ -76,6 +80,52 @@ void AppendBookmarkIds(vector<df::MarkID> const & marks, vector<bookmarks::Id> &
...
+
+
+class BookmarksSearchCallback
+{
+public:
+  using OnResults = BookmarksSearchParams::OnResults;
+
+  template <typename Callback>
+  explicit BookmarksSearchCallback(Callback && onResults)
+    : m_onResults(std::forward<Callback>(onResults))
```
[the remark this answers is not in the document]
Comment (ygorshenin) [reply map/search_api.cpp]: It's strange, but this code does not compile on Linux. Changed to ```OnResults const & onResults```.

**Comment by ygorshenin** [comment]
Done, merge plz.

**Pull request merged**